src/main/java/ch/njol/skript/expressions/ExprNewBannerPattern.java

```diff
@@ -0,0 +1,133 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.bukkitutil.BukkitUtils;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Color;
+import ch.njol.util.Kleenean;
+import org.bukkit.NamespacedKey;
+import org.bukkit.block.banner.Pattern;
+import org.bukkit.block.banner.PatternType;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Stream;
+
+@Name("New Banner Pattern")
+@Description("Create a new banner pattern.")
+@Examples({
+	"set {_pattern} to a new creeper banner pattern colored red",
+	"add {_pattern} to banner patterns of {_banneritem}",
+	"remove {_pattern} from banner patterns of {_banneritem}",
+	"set the 1st banner pattern of block at location(0,0,0) to {_pattern}",
+	"clear the 1st banner pattern of block at location(0,0,0)"
+})
+@Since("INSERT VERSION")
+public class ExprNewBannerPattern extends SimpleExpression<Pattern> {
+
+	private static Object[] patternTypes;
+	private static Map<Integer, Object> patternCorrelation = new HashMap<>();
+
+	static {
+		if (BukkitUtils.registryExists("BANNER_PATTERN")) {
+			try {
+				Class<?> registryClass = Class.forName("org.bukkit.Registry");
+				Object bannerRegistry = registryClass.getField("BANNER_PATTERN").get(null);
+				List<?> registryPatterns = ((Stream<?>) registryClass.getMethod("stream").invoke(bannerRegistry)).toList();
+				patternTypes = registryPatterns.toArray();
+			} catch (ClassNotFoundException | NoSuchFieldException | IllegalAccessException |
+					 InvocationTargetException | NoSuchMethodException e)
+			{
+				throw new RuntimeException(e);
+			}
+
+		} else {
+			try {
+				Class<?> enumClass = Class.forName("org.bukkit.block.banner.PatternType");
+				if (enumClass.isEnum()) {
+					patternTypes = enumClass.getEnumConstants();
+				} else {
+					throw new IllegalStateException("PatternType is neither an enum nor a valid registry.");
+				}
+			} catch (ClassNotFoundException | IllegalStateException e) {
+                throw new RuntimeException(e);
+            }
+        }
+		List<String> patterns = new ArrayList<>();
+		for (int i = 0; i < patternTypes.length; i++) {
+			Object type = patternTypes[i];
+			try {
+				patterns.add("[a] [new] "
+					+ getKey(type)
+					+ " [banner] pattern colo[u]red %*color%"
```

Comment:
```suggestion
				patterns.add("[a] %*color%"
					+ getKey(type)
					+ " [banner] pattern"
```
can use the old pattern for non-literals